### ISSUE TITLE
New `AuthenticationOptions` property in global configuration

### ISSUE THREAD

**Issue by faultcode** [ThreeMammals/Ocelot#842]
Is it possible to use `AuthenticationOptions` in the `GlobalConfiguration`? Like this:

```json
{
  "Routes": [],
  "GlobalConfiguration": {
    "AuthenticationOptions": {
      "AuthenticationProviderKeys": [ "TestKey" ],
      "AllowedScopes": []
    }
  }
}
````
The reason why I would do that is: the `Routes` configuration is coming from a Service Discovery, and I want all of them to be authenticated via IdentityServer with key `TestKey`.

Also, is it possible to add `AllowedScopes` to a service registered to Consul, and then map it to Ocelot Routes dynamically (same as host-names and ports)?

**Comment by faultcode** [comment]
Can anyone please provide me with an answer?

**Comment by harshmaurya** [comment]
Yes, I think this should be an option. I have to explicitly add the AuthenticationOptions in every ReRoute which is a bit of a risky thing. Because, if it is missed in a single reroute by mistake, user would be able to access the data unauthenticated.

**Comment by raman-m** [comment]
@faultcode commented [on May 8, 2019](#issuecomment-490266033)
> Can anyone please provide me with an answer?

I believe the answer to all your Qs is **Yes** because @jlukawska has prepared a nice PR #1215 .
Could you join to code review also please? 😉 

---

The issue has been **accepted** because of ready PR #1215 by @jlukawska

**Comment by raman-m** [comment]
Assigned to @jlukawska
